### PR TITLE
Update 0x08-Testing-Tools.md

### PR DESCRIPTION
**Pull request by Ali-Yazdani**
Adding 3 items in the Android dynamic analysis tools list

Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [ ] Your contribution is written in the 2nd person (e.g. you)
- [ ] Your contribution is written in an active present form for as much as possible.
- [x] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [ ] Your contribution has proper formatted markdown and/or code
- [ ] Any references to website have been formatted as [TEXT](URL “NAME”)
- [ ] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR covers issue #<insert number here>.

### CODE REVIEW

File: Document/0x08-Testing-Tools.md [hunk 1]
```diff
@@ -62,6 +62,9 @@ To perform security testing different tools are available in order to be able to
...
 - [VirtualHook](https://github.com/rk700/VirtualHook "VirtualHook") - VirtualHook is a hooking tool for applications on Android ART(>=5.0). It's based on VirtualApp and therefore does not require root permission to inject hooks.
 - [Xposed Framework](https://forum.xda-developers.com/xposed/xposed-installer-versions-changelog-t2714053 "Xposed Framework") - Xposed framework enables you to modify the system or application aspect and behavior at runtime, without modifying any Android application package(APK) or re-flashing.
+- [jdb](https://docs.oracle.com/javase/7/docs/technotes/tools/windows/jdb.html) - The jdb is a Java Debugger, You can set breakpoint and catch application variables. (jdb worked on jdwp protocol)
+- [Andbug](https://github.com/swdunlop/AndBug) - A Scriptable Android Debugger, AndBug is a debugger targeting the Android platform's Dalvik virtual machine intended for reverse engineers and developers. (andbug worked on jdwp protocol)
+- [Introspy-Android](https://github.com/iSECPartners/Introspy-Android) - Blackbox tool to help understand what an Android application is doing at runtime and assist in the identification of potential security issues.
```
Comment (commjoen) [inline Document/0x08-Testing-Tools.md]: Have you tested Introspy lately? Does it still work on Android oreo/Pie? we see a lot of open issues there...

File: Document/0x08-Testing-Tools.md [hunk 2]
```diff
@@ -62,6 +62,9 @@ To perform security testing different tools are available in order to be able to
 - [logcat-color](https://github.com/marshall/logcat-color "Logcat color") - A colorful and highly configurable alternative to the adb logcat command from the Android SDK.
 - [VirtualHook](https://github.com/rk700/VirtualHook "VirtualHook") - VirtualHook is a hooking tool for applications on Android ART(>=5.0). It's based on VirtualApp and therefore does not require root permission to inject hooks.
 - [Xposed Framework](https://forum.xda-developers.com/xposed/xposed-installer-versions-changelog-t2714053 "Xposed Framework") - Xposed framework enables you to modify the system or application aspect and behavior at runtime, without modifying any Android application package(APK) or re-flashing.
+- [jdb](https://docs.oracle.com/javase/7/docs/technotes/tools/windows/jdb.html) - The jdb is a Java Debugger, You can set breakpoint and catch application variables. (jdb worked on jdwp protocol)
+- [Andbug](https://github.com/swdunlop/AndBug) - A Scriptable Android Debugger, AndBug is a debugger targeting the Android platform's Dalvik virtual machine intended for reverse engineers and developers. (andbug worked on jdwp protocol)
```
Comment (commjoen) [inline Document/0x08-Testing-Tools.md]: Have you tested Andbug on latest Android devices? We see an old code base and many open issues..

Comment (Ali-Yazdani) [reply Document/0x08-Testing-Tools.md]: Hi,
I test it on Android 8.0. It does working correctly.

File: Document/0x08-Testing-Tools.md (hunk 1 above)
Comment (Ali-Yazdani) [reply Document/0x08-Testing-Tools.md]: Introspy worked based on 'Cydia Substrate', so to work we need the Cydia and based on the app website Cydia, Supported Android on versions 2.3 through 4.3. On the mentioned version the Introspy does working correctly.
http://www.cydiasubstrate.com/

Comment (commjoen) [reply Document/0x08-Testing-Tools.md]: I think it is better to test introspy at least and tell the limitations of the android version in the tool description...

File: Document/0x08-Testing-Tools.md [hunk 3]
```diff
@@ -62,6 +62,9 @@ To perform security testing different tools are available in order to be able to
 - [logcat-color](https://github.com/marshall/logcat-color "Logcat color") - A colorful and highly configurable alternative to the adb logcat command from the Android SDK.
 - [VirtualHook](https://github.com/rk700/VirtualHook "VirtualHook") - VirtualHook is a hooking tool for applications on Android ART(>=5.0). It's based on VirtualApp and therefore does not require root permission to inject hooks.
 - [Xposed Framework](https://forum.xda-developers.com/xposed/xposed-installer-versions-changelog-t2714053 "Xposed Framework") - Xposed framework enables you to modify the system or application aspect and behavior at runtime, without modifying any Android application package(APK) or re-flashing.
+- [jdb](https://docs.oracle.com/javase/7/docs/technotes/tools/windows/jdb.html) - The jdb is a Java Debugger, You can set breakpoint and catch application variables. (jdb worked on jdwp protocol)
```
[the remark this answers is not in the document]
Comment (Ali-Yazdani) [reply Document/0x08-Testing-Tools.md]: Thanks, good point

File: Document/0x08-Testing-Tools.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ali-Yazdani) [reply Document/0x08-Testing-Tools.md]: I agreed

[the remark this answers is not in the document]
Comment (sushi2k) [reply Document/0x08-Testing-Tools.md]: Hi @Ali-Yazdani. When do you think you can make an update for the Pull request? Thanks

[the remark this answers is not in the document]
Comment (TheDauntless) [reply Document/0x08-Testing-Tools.md]: (And apply the sam changes to the other pages, of course)

[the remark this answers is not in the document]
Comment (TheDauntless) [reply Document/0x08-Testing-Tools.md]: For JDB, I suggest:

- [jdb](https://docs.oracle.com/javase/7/docs/technotes/tools/windows/jdb.html) - The jdb tool is a Java Debugger which allows you to set breakpoints and print application variables.

[the remark this answers is not in the document]
Comment (sushi2k) [reply Document/0x08-Testing-Tools.md]: Hi @Ali-Yazdani. When do you think you can commit the changes? Thanks

**Comment by Ali-Yazdani** [comment]
Hi,
I will commit the changes the next Saturday.

Best Regards,
Ali

**Comment by TheDauntless** [comment]
Hi @Ali-Yazdani , can you perform the last updates so that we can merge? :)

**Comment by sushi2k** [comment]
Hi @Ali-Yazdani. Please update the PR till 31.5, otherwise we will close it. Thanks

**Comment by Ali-Yazdani** [comment]
Hi @sushi2k 
Please check my PR.
Thanks

File: Document/0x08-Testing-Tools.md [hunk 4]
```diff
@@ -62,9 +62,9 @@ To perform security testing different tools are available in order to be able to
...
-- [Introspy-Android](https://github.com/iSECPartners/Introspy-Android) - Blackbox tool to help understand what an Android application is doing at runtime and assist in the identification of potential security issues.
+- [jdb](https://docs.oracle.com/javase/7/docs/technotes/tools/windows/jdb.html) - The jdb tool is a Java Debugger which allows you to set breakpoints and print application variables. Jdb uses the JDWP protocol.
+- [Andbug](https://github.com/swdunlop/AndBug) - A Scriptable Android Debugger; AndBug is a debugger targeting the Android platform's Dalvik virtual machine intended for reverse engineers and developers. andbug is using the Java Debug Wire Protocol (JDWP
+- [Introspy-Android](https://github.com/iSECPartners/Introspy-Android) - Blackbox tool to help understand what an Android application is doing at runtime and assist in the identification of potential security issues. Limitations: Introspy worked based on 'Cydia Substrate', so to work we need the Cydia and based on the app website Cydia, Supported Android on versions 2.3 through 4.3. On the mentioned version the Introspy does working correctly. (http://www.cydiasubstrate.com/)
```
Comment (TheDauntless) [inline Document/0x08-Testing-Tools.md]: Change to:

- [Introspy-Android](https://github.com/iSECPartners/Introspy-Android) - Blackbox tool to help understand what an Android application is doing at runtime and assist in the identification of potential security issues. Limitations: Introspy is built on 'Cydia Substrate', which is only available for Android versions 2.3 to 4.2 (http://www.cydiasubstrate.com/)

**Comment by Ali-Yazdani** [comment]
Hi,
@TheDauntless well done 
Thanks

**Comment by Ali-Yazdani** [comment]
Dear @commjoen ,
Please check this pull request.
Thanks

File: Document/0x08-Testing-Tools.md (hunk 4 above)
Comment (TheDauntless) [reply Document/0x08-Testing-Tools.md]: You marked it as resolved, but you didn't change it? (The current text also contains some english language errors)

**Comment by sushi2k** [comment]
Let’s merge this for now and resolve some of the issues in another PR. Thanks

**Pull request merged**